server/datastore/mysql/hosts.go

```diff
@@ -840,9 +850,13 @@ func filterHostsByMacOSSettingsStatus(sql string, opt fleet.HostListOptions, par
 	newSQL := ""
 	newParams := []interface{}{}
 
-	if opt.TeamFilter == nil || *opt.TeamFilter == 0 {
-		// add "no team" filter
-		newSQL += ` AND h.team_id IS NULL`
+	if opt.TeamFilter == nil {
+		// TODO(sarah): "All teams" option is not compatible with the macOS settings filter. Should we
+		// coerce to the "no team" option here? Or surface some kind of a usage error in cases where the macOS
+		// settings filter is used without an explicit team id filter? How/where would we
+		// want to surface the error? Maybe follow the approach of processHostFilters at service level?
```

Comment:
I just got off a zoom with Gabe about how we handle "not providing team id" vs "team_id=0" in our endpoints :D We do have some inconsistencies now, as I think for all (most?) MDM-related endpoints, a null team_id means "no team", the only place we have an explicit team_id=0 to mean no team and null to mean all hosts is for those changes that you're adding. I think that's fine for now, it would be a lot of changes to standardize all that 3 days before the freeze, but it's something we should think about standardizing in the future.

My preference (until we normalize at-large) would be to keep it working as it did before, meaning that null would continue to mean "no team" for that filter (unless, of course, that filter can be applied for one of the endpoints that support both "team_id=null" AND "team_id=0"? Is that the case?).

Reply:
Oh I see that before we supported both `nil` and `0` to mean "no team", but now `nil` would be all hosts? Hmmmm I guess yeah in that case maybe it should be part of the `macossettingsfilter` validation on line 846, and ignore this filter if `team_id=nil`? Unless we want to actually support filtering across teams on the macos setting (e.g. if `team_id=nil`, then no condition set on the host's `team_id` so it considers "all hosts").

Reply:
> unless, of course, that filter can be applied for one of the endpoints that support both "team_id=null" AND "team_id=0"? Is that the case?

@mna Yes, I think there are cases where the endpoint supports both. My approach in the PR is to coerce the "no team" result for filters that are not compatible with `team_id=null`.

Under that approach all three of the requests below are equivalent and yield the "no team" result: 
- `GET /hosts?macos_settings=latest&team_id=0` 
- `GET /hosts?macos_settings=latest&team_id=null`
- `GET /hosts?macos_settings=latest`

Is that consistent with what you and @ghernandez345 discussed?

Reply:
Ah ok so in your case, whenever a `macos_settings` is provided, it will necessarily be for a team or no team, never for all teams? That makes sense to me that "no team" is implicit given the nature of that filter. No what we discussed about was a bit different, it was whether we'd change existing mdm-related endpoints to support team_id = 0 to mean "no team" (which I think is a bit risky/late in the sprint to tackle).